src/components/Container.vue

```diff
@@ -34,76 +28,38 @@
 import Filter from "./Filters.vue";
 import Character from "./Character.vue";
 import Search from "./Search.vue";
+import { mapGetters } from "vuex";
 
 export default {
-  props: {
-    msg: String
-  },
   data: function() {
     return {
-      error: null,
-      isLoaded: false,
-      items: [],
-      specie: [],
-      gender: [],
-      filterName: "",
-      filterGender: [],
-      filterSpecies: [],
-      sort: "asc",
-      filteredItems: this.items
+      sort: "asc"
     };
   },
   methods: {
-    updateFilter(event) {
-      this.filterName = event.target.value;
-    },
     sortById(order) {
-      let filteredList =
-        order === "asc"
-          ? this.items.sort((a, b) => (a.id > b.id ? 1 : -1))
-          : this.items.sort((a, b) => (a.id < b.id ? 1 : -1));
-      this.filteredItems = filteredList || this.items;
+      this.$store.commit("sortItems", { order });
+    },
+    filterByGender(event) {
+      this.$store.commit("filterGender", { filterGender: event });
+    },
+    filterBySpecie(event) {
+      this.$store.commit("filterSpecies", { filterSpecies: event });
+    },
+    filterByName() {
+      this.$store.commit("filterName", { filterName: event.target.value });
     }
   },
   components: {
     character: Character,
     search: Search,
     filters: Filter
   },
-  watch: {
-    filterGender() {
-      this.filteredItems = this.items.filter(item => this.filterGender.length > 0 ? this.filterGender.includes(item.gender) : item);
-    },
-    filterSpecies() {
-      this.filteredItems = this.items.filter(item => this.filterSpecies.length > 0 ? this.filterSpecies.includes(item.species) : item);
-    },
-    filterName() {
-      this.filteredItems = this.items.filter(item =>
-        item.name.toLowerCase().includes(this.filterName.toLowerCase())
-      );
-    }
+  created() {
```

Comment:
just to know why created() hook is used instead of mounted()? Any specific reason?